ui/core/components/encounter_picker.ts

```diff
@@ -332,23 +334,51 @@ function addEncounterFieldPickers(rootElem: HTMLElement, encounter: Encounter, s
 		setValue: (eventID: EventID, encounter: Encounter, newValue: number) => {
 			encounter.setDurationVariation(eventID, newValue);
 		},
+		enableWhen: (obj) => { return !encounter.getUseHealth() },
 	});
 
 	if (showExecuteProportion) {
 		new NumberPicker(rootElem, encounter, {
 			label: 'Execute Duration (%)',
 			labelTooltip: 'Percentage of the total encounter duration, for which the targets will be considered to be in execute range (< 20% HP) for the purpose of effects like Warrior Execute or Mage Molten Fury.',
-			changedEvent: (encounter: Encounter) => encounter.executeProportionChangeEmitter,
+			changedEvent: (encounter: Encounter) => encounter.changeEmitter,
 			getValue: (encounter: Encounter) => encounter.getExecuteProportion() * 100,
 			setValue: (eventID: EventID, encounter: Encounter, newValue: number) => {
 				encounter.setExecuteProportion(eventID, newValue / 100);
 			},
+			enableWhen: (obj) => { return !encounter.getUseHealth() },
+		});
+	}
+
+	// Enable for raids and if manually enabled in the advanced options.
+	if (showHealthPicker || encounter.getUseHealth()) {
+		new BooleanPicker<Encounter>(rootElem, encounter, {
+			label: 'Use Health',
+			labelTooltip: 'use a damage limit in place of a duration limit.',
+			changedEvent: (encounter: Encounter) => encounter.changeEmitter,
+			getValue: (encounter: Encounter) => encounter.getUseHealth(),
+			setValue: (eventID: EventID, encounter: Encounter, newValue: boolean) => {
+				encounter.setUseHealth(eventID, newValue);
+			},
+		});
+		let np = new NumberPicker(rootElem, encounter, {
+			label: 'Health',
+			labelTooltip: 'If picked, ignores duration setting and instead fights until the raid kills the primary target. Effectively a damage limit instead of time limit.',
+			changedEvent: (encounter: Encounter) => encounter.changeEmitter,
+			getValue: (encounter: Encounter) => encounter.getHealth(),
+			setValue: (eventID: EventID, encounter: Encounter, newValue: number) => {
+				encounter.setHealth(eventID, newValue);
+			},
+			enableWhen: (obj) => { return encounter.getUseHealth() },
```

Comment:
This seems unnecessary, we already have a health stat for each target. Use the sum of the health of all the targets as the health value.

Reply:
This needs to be here so that you can edit the health without having to open the advanced pane

Reply:
Oh, I would say the whole 'use health' option should be in the advanced pane.